### PR TITLE
Bulk of Library Work Completed

### PR DESCRIPTION
**Pull request by melissagirl**
This is the initial work for Circuit Python and is based of of the Adafruit RA8875 Arduino Library, but with some structural changes so that it's even easier to use. Many of the function names and parameters are intended to coincide with the GFX library for easy of porting code over. I have also ran this through PyLint and the only things coming up are import-errors and I'm sure that has to do with my setup as it functions perfectly.

There are 2 examples. One goes through all of the hardware accelerated drawing function as well as displaying text and updating it. The other one will decode and draw a bitmap onto the screen using the fastest drawing methods available.

### CODE REVIEW

**Comment by melissagirl** [comment]
Good catch. They probably didn’t copy over because they were hidden.

—

Melissa LeBlanc-Williams

Check out my YouTube Channel at https://www.youtube.com/channel/UCmDzlRL-Wkg3HCBlF58A4cw

Follow me on Twitter at https://twitter.com/makermelissa

**Comment by melissagirl** [comment]
Ok, I re-added the files. to I need to create a new PR?

—

Melissa LeBlanc-Williams

Check out my YouTube Channel at https://www.youtube.com/channel/UCmDzlRL-Wkg3HCBlF58A4cw

Follow me on Twitter at https://twitter.com/makermelissa

**Comment by kattni** [comment]
Perfect, thank you! Travis is now running on it and has failed. (As it does for all of us!) 

We've moved to Discord for now for more immediate discussion.

File: adafruit_ra8875/ra8875.py [hunk 1]
```diff
@@ -0,0 +1,617 @@
...
+    """Convert red, green and blue values (0-255) into a 16-bit 565 encoding."""
+    try:
+        r, g, b = r  # see if the first var is a tuple/list
+    except TypeError:
+        pass
+    return (r & 0xf8) << 8 | (g & 0xfc) << 3 | b >> 3
+#pylint: enable-msg=invalid-name
+
+class RA8875_Device(object):
+    """Set Initial Variables"""
```
Comment (sommersoft) [inline adafruit_ra8875/ra8875.py]: Can you add parameter descriptions for the class, so they're available in the documentation? 
Example:
```
"""
Writes the specified 'payload' to the sensor
Returns the results of the write attempt.

:param bytearray payload: The byte array to write to the sensor
:param rxlen: (optional) The numbers of bytes to read back (default=0)
"""
```
Requesting this for each class, as well as each public function (private function docs aren't built by default, though can be forced).

File: docs/api.rst [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+.. If you created a package, create one automodule per module in the package.
+
+.. If your library file(s) are nested in a directory (e.g. /adafruit_foo/foo.py)
+.. use this format as the module name: "adafruit_foo.foo"
+
+.. automodule:: adafruit_ra8875.ra8875
+   :members:
+
+.. automodule:: adafruit_ra8875.registers
+   :members:
```
Comment (sommersoft) [inline docs/api.rst]: `registers` seems to be for internal use only? If they aren't going to be exposed to the user, I don't really see a need to list them in the documentation. _May_ add a layer of confusion. I can definitely be overruled on this, though. 😄

**Comment by melissagirl** [comment]
Thanks @sommersoft. I’ll work on that today.

—

Melissa LeBlanc-Williams

Check out my YouTube Channel at https://youtube.com/makermelissaslab

Follow me on Twitter at https://twitter.com/makermelissa

File: docs/api.rst (hunk 2 above)
Comment (melissagirl) [reply docs/api.rst]: That is correct. Internal use only. I just included them in a separate file because they cluttered up the code.

Comment (melissagirl) [reply docs/api.rst]: Oh I get it, should I remove the automodule for that then?

File: adafruit_ra8875/ra8875.py (hunk 1 above)
Comment (melissagirl) [reply adafruit_ra8875/ra8875.py]: Ok, I added parameters for all functions. I used http://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#info-field-lists as a reference.

File: docs/api.rst (hunk 2 above)
Comment (melissagirl) [reply docs/api.rst]: I went ahead and removed the automodule.

**Comment by melissagirl** [comment]
I've gone ahead and made the above changes. It passed on Travis just fine and I don't think there's anything left to do on my end.

**Comment by melissagirl** [comment]
@tannewt Thanks! There are most likely other hardware accelerated graphics chips like this, but I'm not aware of anything specific.

**Pull request merged**